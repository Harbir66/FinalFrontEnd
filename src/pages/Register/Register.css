.register{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    height: 100vh;
    width: 100vw;
    background-color: #EFF1FF;
}

.register-image-container{
    padding: 10vh  10vh 20vh 10vh;
    width: 60vw;
    height: 70vh;
    background-color: #EFF1FF;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.register-image-container > div{
    text-align: start;
    color: #4F4F4F;
    margin: 0;
}

.register-image-container > div > h1{
    align-self: flex-start;
}

.register-image{
    align-self: flex-end;
    width: 80%;
    background-color: #e7baba;
    background-color: transparent;
}

.register-form{
    padding-top: 20vh;
    padding-bottom: 10vh;
    width: 40vw;
    height: 70vh;
    background-color: #272727;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
}

.register-form > h1{
    color: #EFF1FF;
    margin: 0;
}

.register-form > form{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 30vw;
}

.register-form > form > label{
    text-align: start;
    color: #707070;
    margin-bottom: 2px;
    width: 100%;
}
.register-form > form > input{
    width: 95%;
    height: 4vh;
    margin-bottom: 2vh;
    border: none;
    border-radius: 5px;
    padding: 0 1vw;
}

.register-form > form > button{
    width: 50%;
    height: 4vh;
    border: none;
    border-radius: 5px;
    background: linear-gradient(70deg, #9A6EFF 0%, #643EFF 100%);
    color: #EFF1FF;
    font-weight: bold;
    font-size: smaller;
    cursor: pointer;
}

.register-button > button{
    background: none;
    border: none;
    color: #707070;
    font-weight: bold;
    font-size: smaller;
    cursor: pointer;
    text-decoration: underline;
    margin-top: 1rem;
}

@media screen and (max-width: 768px) {
    .register-image-container{
        width: 50vw;
        padding: 10vh 0 0 0;
    }
    .register-image{
        width: 100%;
    }
    .register-form{
        width: 50vw;
    }
}
    
@media screen and (max-width: 525px) {
    .register{
        flex-direction: column;
    }
    .register-image-container{
        width: 100vw;
        height: 50vh;
        padding: 0;
    }
    .register-form{
        width: 100vw;
        height: 50vh;
    }
}
    
