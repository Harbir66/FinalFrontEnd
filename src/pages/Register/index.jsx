import React from 'react';
import axios from 'axios';
import './Register.css';
import { useNavigate } from 'react-router-dom';
import registerImage from '../../assets/undraw-upload-re-pasx@3x.png';
import { ERROR_ROUTE } from '../../constants/routes';

function Register() {
  const [email, setEmail] = React.useState('');
  const [password, setPassword] = React.useState('');
  const [confirmPassword, setConfirmPassword] = React.useState('');
  const [text, setText] = React.useState('Register');
  const navigate = useNavigate();
  const handleClick = () => {
    navigate('/login');
  };
  const handleSubmit = async (e) => {
    if (password !== confirmPassword) {
      alert('Password and Confirm Password should be same. TRY AGAIN !!');
      return;
    }
    e.preventDefault();
    const data = {
      email,
      password,
    };
    try {
      const requestDetails = {
        method: 'POST',
        url: 'http://localhost:3010/api/register',
        data,
      };
      const response = await axios(requestDetails);
      if (response.status === 201) {
        setText('Registered !!');
        setTimeout(() => {
          navigate('/login');
        }, 1000);
      }
    } catch (error) {
      if (navigate) {
        const errorCode = error.response?.status;
        if (errorCode) {
          navigate(`${ERROR_ROUTE}/${errorCode}`);
        } else {
          navigate(`${ERROR_ROUTE}`);
        }
      }
    }
  };
  return (
    <div className="register">
      <div className="register-image-container">
        <div>
          <h1>Design APIs fast,</h1>
          <h1>Manage content easily.</h1>
        </div>
        <img
          className="register-image"
          src={registerImage}
          alt="register-art"
        />
      </div>
      <div className="register-form">
        <h1>Register for CMS+ account</h1>
        <form onSubmit={handleSubmit}>
          <label htmlFor="email">Email</label>
          <input
            type="email"
            id="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <label htmlFor="password">Password</label>
          <input
            type="password"
            id="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <label htmlFor="confirmPassword">Confirm Password</label>
          <input
            type="password"
            id="confirmPassword"
            value={confirmPassword}
            onChange={(e) => setConfirmPassword(e.target.value)}
          />
          <button type="submit">{text}</button>
          <span className="register-button">
            <button onClick={handleClick} type="button">
              Already a User??
            </button>
          </span>
        </form>
      </div>
    </div>
  );
}

export default Register;
