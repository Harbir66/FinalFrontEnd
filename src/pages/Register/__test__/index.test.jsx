import { render } from '@testing-library/react';
import Register from '../index';

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

describe('Register', () => {
  test('Register should render correctly', () => {
    const { asFragment } = render(<Register />);
    expect(asFragment()).toMatchSnapshot();
  });
});
