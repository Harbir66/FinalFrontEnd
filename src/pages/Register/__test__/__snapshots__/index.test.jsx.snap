// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Register should render correctly 1`] = `
<DocumentFragment>
  <div
    class="register"
  >
    <div
      class="register-image-container"
    >
      <div>
        <h1>
          Design APIs fast,
        </h1>
        <h1>
          Manage content easily.
        </h1>
      </div>
      <img
        alt="register-art"
        class="register-image"
        src="undraw-upload-re-pasx@3x.png"
      />
    </div>
    <div
      class="register-form"
    >
      <h1>
        Register for CMS+ account
      </h1>
      <form>
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          type="password"
          value=""
        />
        <label
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <input
          id="confirmPassword"
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Register
        </button>
        <span
          class="register-button"
        >
          <button
            type="button"
          >
            Already a User??
          </button>
        </span>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
